File: backend-mealsonwheels/src/main/java/com/merrymeal/mealsonwheels/dto copy/mealDTOs/DishDTO.java
package com.merrymeal.mealsonwheels_backend.dto.mealDTOs;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class DishDTO {

    private Long id;

    @NotBlank(message = "Dish name is required")
    @Size(max = 100)
    private String dishName;

    private byte[] dishPhoto;

    @Size(max = 500)
    private String dishDesc;

    private Long partnerId;
    private Long mealId;
    private Long menuId;

    private String dishType;
    private String dishDietary;

    private LocalDateTime dishCreatedDate;
}

File: frontend-mealsonwheels/src/types/meal/DishDTO.ts
// src/types/meal/DishDTO.ts

export interface DishDTO {
  id: number;
  dishName: string;
  dishPhoto: Uint8Array | null; // Representing byte[] in JS
  dishDesc: string;
  partnerId: number;
  mealId: number;
  menuId: number;
  dishType: string;
  dishDietary: string;
  dishCreatedDate: string; // ISO string (LocalDateTime)
}
